### PR TITLE
E_CLASSROOM-166 [FE] [Student] [Integration] Recent Activity Function…

### PR DESCRIPTION
**Pull request by catampongan**
### Issue Link
https://framgiaph.backlog.com/view/E_CLASSROOM-166

### Definition of Done
- [x] Should return 4 data of user's most recent taken quizzes.

### Commands to Run

N/A

#### Main note
- (Therese) Please checkout from the Backend PR I created before testing because I made changes there to cater the data needed for the recent quizzes tables. 
- Link to the BE PR: https://github.com/framgia/sph-classroom-els-be/pull/36
#### Pages Affected
- http://localhost:3003/

### Scenarios/ Test Cases
- [ ] Should return 4 data of user's most recent taken quizzes.
#### User Interface
- N/A
#### User Experience 
- N/A
#### Functionality
- N/A
### Screenshots
image

### CODE REVIEW

**Comment by geneno99** [comment]
### Comment 
- [x] Put text in new user because by default it has no recent quizzes taken
new user no recent activity

- [ ] If the user retake specific quiz, in the dashboard it create new card. I think it must `not` create new card since in the card it has text display `Attempts` section. Please confirm 🙇🏼 
recent quiz 2

----- 
- Test time: 0.5 hr

**Comment by geneno99** [comment]
## Approved PR #53 
- [x] Able to display user recent quiz taken in the dashboard
- [x] Maximum of 4 recent quizzes is being displayed
- [x] Able display categories taken in categories learned section

4 recent quizzes taken

## Note
- Second comment is disregard for now.

--------
- Total Test time: 0.75 hr
- Total Iterations: 1
- Total Comments: 1

File: src/pages/student/main/Dashboard/components/Recent/index.module.css [hunk 1]
```diff
@@ -1,56 +1,58 @@
-.bg{
-    display: flex;
-    justify-content: space-between;
-    margin-bottom: 33px;
+.bg {
+  display: flex;
+  margin-bottom: 40px;
```
Comment (RunstyleX) [inline src/pages/student/main/Dashboard/components/Recent/index.module.css]: please use 4 spaces for css files

File: src/pages/student/main/Dashboard/index.module.css [hunk 2]
```diff
@@ -1,50 +1,59 @@
...
+  display: inline-block;
+  font-weight: 500;
+  font-size: 18px;
 }
 
-.titleText{
-    display: inline-block;
-    font-weight: 500;
-    font-size: 18px;
+.btnVeiw {
```
Comment (RunstyleX) [inline src/pages/student/main/Dashboard/index.module.css]:
```suggestion
.btnView {
```

File: src/pages/student/main/Dashboard/index.js [hunk 3]
```diff
@@ -57,17 +62,17 @@ function Dashboard() {
...
+                      return (
+                        <Recent
+                          recentQuizzes={recentQuizzes}
+                          quizzes={getAllQuizzesTakenForEveryRecentQuiz(
+                            recentQuizzes.quiz_id
+                          )}
+                          key={idx}
+                        />
+                      );
+                    })}
```
Comment (RunstyleX) [inline src/pages/student/main/Dashboard/index.js]: No need to indent so far

File: src/pages/student/main/Dashboard/index.js [hunk 4]
```diff
@@ -42,11 +56,28 @@ function Dashboard() {
...
+              <Recent
+                recentQuizzes={recentQuizzes}
+                quizzes={getAllQuizzesTakenForEveryRecentQuiz(
+                  recentQuizzes.quiz_id
+                )}
+                key={idx}
+              />
+            );
+          })}
       </div>
```
Comment (RunstyleX) [inline src/pages/student/main/Dashboard/index.js]:
```suggestion
            );
          })
        }
      </div>
```

**Pull request merged**